### DISCUSSION TITLE
How to change the response type of /openapi.json

### DISCUSSION THREAD

**Discussion by captaincoordinates** [fastapi/fastapi#6760] [Questions] [answered]
There appears to be an [active discussion](https://github.com/OAI/OpenAPI-Specification/issues/110) around the preferred MIME type for OpenAPI documentation. FastAPI provides /openapi.json with the `application/json` MIME type.

I am attempting to use FastAPI to implement the OGC API - Features - Part 1: Core 1.0 spec which requires OpenAPI documentation be provided with MIME type application/vnd.oai.openapi+json;version=3.0 (see [here](http://docs.ogc.org/is/17-069r3/17-069r3.html#_basic_requirements)).

Is it possible to change the response type of /openapi.json within FastAPI? Looking through the documentation and searching around the internet I don't see any way to do this.

**Comment by waynerv** [comment]
Thanks for the flexiable of FastAPI, this is pretty much easy. You just need  create a custom `Response` class, and ues it as `response_class` in route:
```python
from typing import List, Optional

from fastapi import FastAPI
from pydantic import BaseModel
from starlette.responses import JSONResponse

app = FastAPI()


class Item(BaseModel):
    name: str
    description: Optional[str] = None
    price: float
    tax: Optional[float] = None
    tags: List[str] = []


class VndResponse(JSONResponse):
    media_type = 'application/vnd.oai.openapi+json;version=3.0'


@app.post("/items/", response_model=Item, response_class=VndResponse)
async def create_item(item: Item):
    return item
```
![image](https://user-images.githubusercontent.com/39515546/113239062-806cb380-92dc-11eb-9f23-b543240cdd61.png)

**Comment by juntatalor** [comment]
@captaincoordinates Hi!
The openapi handler is declared inside the `FastAPI.setup()` as an inner function that returns `JSONresponse`. I'm afraid that the only one way is to subclass FastAPI and write your own `setup` implementation.

**Comment by captaincoordinates** [comment, marked as answer]
@juntatalor thanks for the info. Anyone else who's looking for this I found the described method [here](https://github.com/tiangolo/fastapi/blob/10397ddc30ef9bb26e9247f04fe20ce3045081c4/fastapi/applications.py#L141).

**Comment by tiangolo** [comment]
Thanks for the help here everyone! 👏  🙇 

Thanks for reporting back and closing the issue @captaincoordinates  👍

> Sorry for the long delay! 🙈  I wanted to personally address each issue/PR and they piled up through time, but now I'm checking each one in order.
